packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -259,14 +259,15 @@ class AndroidDevice extends Device {
     return true;
   }
 
-  Future<int> _forwardPort(String service, int devicePort, int port) async {
+  Future<Uri> _forwardPort(String service, Uri deviceUri, int port) async {
```

Comment:
This method is named '_forwardPort' and it should continue to take ports and return a port.